artisatomic/readmonsdata.py

```diff
@@ -0,0 +1,198 @@
+    )
+
+    with ziparchive_outglv.open(datafilename_energylevels) as datafile_energylevels:
+        energy_levels1000percm, j_arr = np.loadtxt(datafile_energylevels, unpack=True, delimiter=",")
+    artisatomic.log_and_print(flog, f"levels: {len(energy_levels1000percm)}")
+
+    energiesabovegsinpercm = energy_levels1000percm * 1000
+
+    g_arr = 2 * j_arr + 1
+
+    # Sort table by energy levels
+    dfenergylevels = pd.DataFrame.from_dict({"energiesabovegsinpercm": energiesabovegsinpercm, "g": g_arr})
+    dfenergylevels = dfenergylevels.sort_values("energiesabovegsinpercm")
+
+    energiesabovegsinpercm = dfenergylevels["energiesabovegsinpercm"].values
+    g_arr = dfenergylevels["g"].values
+
+    parity = None  # Only E1 so always allowed transitions.
+    energy_levels = [None]
+
+    for levelindex, (g, energyabovegsinpercm) in enumerate(zip(g_arr, energiesabovegsinpercm, strict=False)):
+        energy_levels.append(
+            EnergyLevel(levelname=str(levelindex), parity=parity, g=g, energyabovegsinpercm=energyabovegsinpercm)
+        )
+
+    # Read next file
+    ziparchive_outggf = zipfile.ZipFile(datafilepath / "outggf_Ln_V--VII.zip", "r")
+    datafilename_transitions = (
+        f"outggf_Ln_V--VII/outggf_sorted_{artisatomic.elsymbols[atomic_number]}_{artisatomic.roman_numerals[ion_stage]}"
+    )
+
+    with ziparchive_outggf.open(datafilename_transitions) as datafile_transitions:
+        transition_wavelength_A, energy_levels_lower_1000percm, oscillator_strength = np.loadtxt(
+            datafile_transitions, unpack=True, delimiter=","
+        )
+    artisatomic.log_and_print(flog, f"transitions: {len(energy_levels_lower_1000percm)}")
+
+    energy_levels_lower_percm = energy_levels_lower_1000percm * 1000
+
+    # Get index of lower level of transition
+    lowerlevel = np.array(
+        [
+            (
+                np.abs(
+                    energiesabovegsinpercm - energylevellower
+                )  # get closest energy in energy level array to lower level
+            ).argmin()  # then get the index with argmin()
+            for energylevellower in energy_levels_lower_percm
+        ]
+    )
+
+    ionization_energy_in_ev_nist = artisatomic.get_nist_ionization_energies_ev()[(atomic_number, ion_stage)]
+
+    # get energy of upper level of transition
+    energy_levels_lower_ev = energy_levels_lower_percm * hc_in_ev_cm
+    transitionenergyev = hc_in_ev_angstrom / transition_wavelength_A
+    ionization_energy_in_ev = max(transitionenergyev)
+    artisatomic.log_and_print(
+        flog, f"ionization energy: {ionization_energy_in_ev} eV (NIST: {ionization_energy_in_ev_nist} eV)"
+    )
+
+    # If ionisation potential in data does not match NIST to within 1 decimal place
+    # then use NIST instead (probably more accurate?)
+    if round(ionization_energy_in_ev, 1) != round(ionization_energy_in_ev_nist, 1):
```

Comment:
This is a bit of an odd condition (floating-point equality checks should generally be avoided). I think it's the same as abs(ionization_energy_in_ev - ionization_energy_in_ev_nist) > 0.1 ?

Reply:
Sure, should be the same thing